### DISCUSSION TITLE
ESP32 always crashes on boot

### DISCUSSION THREAD

**Discussion by brokenbrigde9000** [micropython/micropython#11063] [ESP32]
I'm running micropython on an ESP32 module and it always crashes on boot, at least the first time. When connected to a computer (I'm using PyCharm with the micropython login) there is always this 'keyboard interrupt' error or some error on a random line in the code. When doing a soft reboot (ctrl+D) it will boot like it should and run the code perfectly fine. When powered as standalone device, it will not boot the first time either, only after pressing the reset button the ESP32 runs the code like it should.

This happens on all my ESP32 units and no matter what code I'm running.

I'm currently using the hardware watchdog to get the ESP32 to reset so it autmatically performs a 2nd attempt to boot. This always works. I'm sure there is something I am missing because this looks like a very dirty solution. How do I get it to boot at once?

**Comment by robert-hh** [comment]
Which ESP32 units do you used, and which firmware version?

**Reply by brokenbrigde9000** [reply]
I'm using the DFRobot Firebeetle 2 ESP32-E and downloaded the firmware v1.19.1 (2022-06-18) to it with Thonny. I will try a capacitor between EN and GND and see if it solves the problem.

**Reply by robert-hh** [reply]
You can also run the nightly builds. They are pretty stable, and since v1.19.1 there were quite a few improvements.

**Comment by robert-hh** [comment]
Besides that, it looks as either a power problem or interference into the EN pin. You could try to put a 1µF to 10µF capacitor between EN and GND.

**Reply by brokenbrigde9000** [reply]
Unfortunately a capacitor between EN and GND does not make a difference, the same goes for using a newer (nightly) version. Any other ideas that might work?

**Comment by bulletmark** [comment]
Long ago I had a similar problem (won't bother describing the details here) but I fixed it by disabling brownout detection, i.e. set `CONFIG_ESP32_BROWNOUT_DET=n` in my `sdkconfig.board`.
